test/prototype/test_bitpacking.py

```diff
@@ -1,70 +1,45 @@
 import torch
-from torchao.prototype.common.bitpacking import pack, unpack
+from torchao.prototype.common.bitpacking import pack, unpack, dtype_to_bits
 import pytest
 from torch.utils._triton import has_triton
 from torchao.quantization.utils import TORCH_VERSION_AFTER_2_4
 
 if not TORCH_VERSION_AFTER_2_4:
     pytest.skip("Unsupported PyTorch version", allow_module_level=True)
 
-def test_uint4_to_uint8_CPU():
-    test_tensor = torch.randint(0, 15, (4, 4), dtype=torch.uint8)
-    packed = pack(test_tensor, 8, 4, device='cpu')
-    unpacked = unpack(packed, 4, device='cpu')
-    unpadded = unpacked[:test_tensor.shape[0], ...]
-    assert(unpadded.allclose(test_tensor))
+def test_trinary_to_uint8_CPU():
+    test_tensor = torch.randint(-1, 1, (4, 4, 4), dtype=torch.int32)
+    for i in range(len(test_tensor.shape)):
+        packed = pack(test_tensor, "trinary", dimension = i, container_dtype = torch.uint8, device='cpu')
+        unpacked = unpack(packed, "trinary", dimension = i, device='cpu')
+        assert(unpacked.to(torch.int32).allclose(test_tensor))
 
-def test_uint3_to_int16_col_wise_cpu():
-    test_tensor = torch.randint(0, 7, (8, 5), dtype=torch.int16)
-    packed = pack(test_tensor,16, 3, False, device='cpu')
-    unpacked = unpack(packed, 3, False, device='cpu')
-    unpadded = unpacked[:test_tensor.shape[0], ...]
-    assert(unpadded.allclose(test_tensor))
-    
+def test_to_uint8_CPU():
+    for dtype in {torch.uint2, torch.uint3, torch.uint4, torch.uint5, torch.uint6, torch.uint7}:
```

Comment:
favor pytest parametrize that way it'll be easier to identify the failing test in CI